### PR TITLE
Dockerization

### PR DESCRIPTION
**Pull request by dogversioning**
This PR adds the following feature:

- Dockerfiles for building transformer images
  - These are up on dockerhub for the latest version
  - Dockerhub has some light instructions for running containers from the images.

### CODE REVIEW

**Comment by dogversioning** [comment]
n.b. - The dockerfiles, at the moment, do not contain caches of the huggingface models, mostly to spread this out over two commits. The main question for later is one of preference; would you rather:
- have one image type per runnable model (smaller individual images, but more maintenance)
- import configurations from the rest modules (brittle but DRYer)
- have a seperate python file just for managing installing models during docker builds (less intrusive but forgettable)

File: setup.cfg [hunk 1]
```diff
@@ -31,7 +31,7 @@ install_requires =
     nltk~=3.5
     numpy ~=1.21.5; python_version<'3.8'
     numpy ~=1.23.3; python_version>='3.8'
-    pandas~=1.3.5
+    pandas~=1.5.2
```
Comment (dogversioning) [inline setup.cfg:34]: so - i guess this is as good a place as any to ask this - I ran into images on the CUDA image with this vintage of pandas, but I see that that's breaking the 3.7 release. Is it safe to specify a range of 1.3.5-1.5.2?

Comment (angus-lherrou) [reply setup.cfg:34]: We could do what I did for numpy and specify a different version for 3.7 and 3.8+. What was the issue with 1.3.5?

**Comment by angus-lherrou** [comment]
> n.b. - The docker images, at the moment, do not contain caches of the huggingface models, mostly to spread this out over two commits. The main question for later is one of preference; would you rather:
> 
>     * have one image type per runnable model (smaller individual images, but more maintenance)
> 
>     * have one image for all models (probably adds a few gigs)
My immediate preference is for one image per runnable model, probably with a base image with the common stuff, but I'll defer to Tim as he's the one who maintains the runnable models.

> and also, would you rather:
> 
>     * import configurations from the rest modules (brittle but DRYer)
> 
>     * have a seperate python file just for managing installing models during docker builds (less intrusive but forgettable)
Could you elaborate on this more concretely? I don't have an immediate preference.

**Comment by dogversioning** [comment]
> 
> > and also, would you rather:
> > 
> > * import configurations from the rest modules (brittle but DRYer)
> > * have a seperate python file just for managing installing models during docker builds (less intrusive but forgettable)
> 
> Could you elaborate on this more concretely? I don't have an immediate preference.

So, in order to get the huggingface transformers into the images, i'm basically looking to mimic the calls to to `from_pretrained` and their supporting infrastructure in `src.cnlpt.api.cnlp_rest.initialize_cnlpt_model` during the docker build. So for each model, I need to go grab arguments from the associated rest support module - most importantly, `model_name`.

So - in the first case, for `negation_rest` as an example, I'd explictly import `model_name` from there, and I might need to do something like make the cuda/batch size arguments more explictly stored in variables in that module so they could be imported as well, and run that during the build for the negation image, and basically do that once for each model.

The other option is to have that code not depend directly on the model code, and explicitly set up a replica set of calls, maintaining `model_name` params in there as well, and managing optional arguments as needed.

Does that make sense? I don't really have a strong preference myself - I think in a vacuum I would probably choose the second option and expect that things would occasionally break, but I can see the other case as well.

File: setup.cfg (hunk 1 above)
Comment (dogversioning) [reply setup.cfg:34]: well, it looks like i can't replicate the issue now, so i'll go ahead and reset this. 💻 🔥

**Comment by angus-lherrou** [comment]
I'll defer to @tmills since that's his code.

**Comment by tmills** [comment]
I'm ok with one image per runnable model.

**Comment by tmills** [comment]
For the second issue, if I understand correctly, I agree with the second approach: Keep a separate set of configs for the image creation. There is nothing magic about the hardcoded configs, and in fact maybe they should be factored out somehow.

**Comment by tmills** [comment]
Any other concerns to address before merging?

**Comment by dogversioning** [comment]
> Any other concerns to address before merging?

Nope, I'm good with that - I'll add multiple images/models in a followup PR.

**Pull request merged**